syntax = "proto3";

package backup;

import "google/protobuf/empty.proto";

service BackupService {
  rpc ResetKey(stream ResetKeyRequest) returns (google.protobuf.Empty) {}
  rpc SendLog(SendLogRequest) returns (google.protobuf.Empty) {}
  rpc PullBackupKey(PullBackupKeyRequest) returns (PullBackupKeyResponse) {}
  rpc PullCompaction(PullCompactionRequest) returns (stream PullCompactionResponse) {}
}

// ResetKey

message ResetKeyRequest {
  string userId = 1;
  oneof data {
    bytes newKey = 2;
    bytes compactionChunk = 3;
  }
}

// SendLog

message SendLogRequest {
  string userId = 1;
  bytes data = 2;
}

// PullBackupKey

message PullBackupKeyRequest {
  string userId = 1;
  bytes pakeKey = 2;
}

message PullBackupKeyResponse {
  bytes encryptedBackupKey = 1;
}

// PullCompaction

message PullCompactionRequest {
  string userId = 1;
  bytes message = 2;
}

message PullCompactionResponse {
  oneof data {
    bytes compactionChunk = 1;
    bytes logChunk = 2;
  }
}
